controllers/helpers.go

```diff
@@ -228,21 +228,32 @@ func userAssignedIdentityCloudProviderConfig(d azure.ClusterScoper, identityID s
 }
 
 func newCloudProviderConfig(d azure.ClusterScoper) (controlPlaneConfig *CloudProviderConfig, workerConfig *CloudProviderConfig) {
+	routeTableName := ""
+	securityGroupName := ""
+	subnetName := ""
+	subnets := d.Subnets()
+	for _, subnet := range subnets {
+		if subnet.Role == infrav1.SubnetNode {
+			routeTableName = subnet.RouteTable.Name
+			securityGroupName = subnet.SecurityGroup.Name
+			subnetName = subnet.Name
+		}
+	}
```

Comment:
I wasn't sure about how to solve this. I'm open to suggestions on how to improve it.

Reply:
Need to check where cloudprovider uses the subnets, if each config needs to have its own subnet name it would make things more complicated. @alexeldeib had said in #664 
> cloud provider doesn't care about subnets except for ILB I think

Reply:
regardless, I would suggest moving this logic to a helper function

Reply:
I moved it to a helper function, PTAL.